torchdynamo/variables/dicts.py

```diff
@@ -14,18 +14,22 @@
 
 
 class ConstDictVariable(VariableTracker):
-    def __init__(self, items, **kwargs):
+    def __init__(self, items, dict_cls=collections.OrderedDict, **kwargs):
         super(ConstDictVariable, self).__init__(**kwargs)
+        self.dict_cls = dict_cls
         if not isinstance(items, collections.OrderedDict):
             assert isinstance(items, dict)
+            self.dict_cls = dict
```

Comment:
Agree to just wrap the object and keep the original dict type, either ```dict``` or ```OrderedDict```, depends on user code's type. Actually right now the sorting in L23 may break the original dict's order. Python ```dict``` keep the insert order rather than sorted order.
Another thing I want to discuss is actually the subclass ```DataClassVariable``` has a ```user_cls``` variable which is used as similar purpose. Do you think we can refactor to move this variable to the parent class? Not sure if the name is confusing, but looks more elegant.

Reply:
I think we should remove OrderedDict entirely from ConstDictVariable, you are correct it gets the order wrong.

Yeah we could move the `user_cls` handling up to this class